raiden/network/proxies/user_deposit.py

```diff
@@ -92,6 +117,151 @@ def whole_balance_limit(self, block_identifier: BlockSpecification) -> TokenAmou
             )
         )
 
+    def init(
+        self,
+        monitoring_service_address: MonitoringServiceAddress,
+        one_to_n_address: OneToNAddress,
+        given_block_identifier: BlockSpecification,
+    ) -> None:
+        """ Initialize the UserDeposit contract with MS and OneToN addresses """
+        log_details = {
+            "monitoring_service_address": to_checksum_address(monitoring_service_address),
+            "one_to_n_address": to_checksum_address(one_to_n_address),
+        }
+
+        check_address_has_code(
+            client=self.client,
+            address=Address(monitoring_service_address),
+            contract_name=CONTRACT_MONITORING_SERVICE,
+            expected_code=decode_hex(
+                self.contract_manager.get_runtime_hexcode(CONTRACT_MONITORING_SERVICE)
+            ),
+        )
+        check_address_has_code(
+            client=self.client,
+            address=Address(one_to_n_address),
+            contract_name=CONTRACT_ONE_TO_N,
+            expected_code=decode_hex(self.contract_manager.get_runtime_hexcode(CONTRACT_ONE_TO_N)),
+        )
+        try:
+            existing_monitoring_service_address = self.monitoring_service_address(
+                block_identifier=given_block_identifier
+            )
+            existing_one_to_n_address = self.one_to_n_address(
+                block_identifier=given_block_identifier
+            )
+        except ValueError:
+            pass
```

Comment:
I don't think we should silently ignore ValueErrors during the initialization. We should confirm that the contracts are initialized correctly before using them.

Reply:
That's the pattern we use for pruned blocks as web3 would raise a `ValueError` when the block at `given_block_identifier` was pruned. In this case, we skip the precondition checks.

Reply:
I was in the wrong part of the code in my head. I thought we were skipping checks for correctly initialized contract addresses, which we should not do.